[feature-prompts] Standardize RBAC logic for new prompts

AlanCoding opened the pull request:
##### SUMMARY
This is mainly about doing DRY edits, but I hope there are a few bugs that will get swept up in this too.

##### ISSUE TYPE
 - Bug, Docs Fix or other nominal change

##### COMPONENT NAME
 - API

AlanCoding committed:
DRY edits to access classes for new prompts

Remove if-not-data conditional from WFJTnode.can_change
  these are cannonical for can_add, but this looks like a bug

Change JTaccess.can_unattach to call same method in super()
  previously called can_attach, which is problematic

Better consolidate launch config m2m related checks

john-westcott-iv commented on awx/main/access.py:
```diff
@@ -328,6 +336,8 @@ def user_has_resource_access(resource):
             role = getattr(resource, role_field, None)
             if role is None:
                 # Handle special case where resource does not have direct roles
+                if role_field == 'read_role':
+                    return self.user.can_access(type(resource), 'read', resource)
```
Why not just add an entry to the access_method_type field below?

AlanCoding replied:
Because the method signature is different, b/c the "read" check does not take the `data` arg or kwarg.

john-westcott-iv commented on awx/main/access.py:
```diff
@@ -1833,184 +1839,60 @@ class JobLaunchConfigAccess(BaseAccess):
-            return False
-        return self.check_related('inventory', Inventory, data, role_field='use_role')
+                    # If given model objects, only use the primary key from them
+                    sub_obj_pks = [cred.pk for cred in data[relationship]]
```
Do we want to remove references to cred in here since it can really be any of the M2M items?

AlanCoding replied:
yes, thanks

AlanCoding commented on awx/main/access.py:
```diff
@@ -1833,184 +1841,63 @@ class JobLaunchConfigAccess(BaseAccess):
      - use role to many-related credentials (if any present)
-     - use role to Execution Environment (if present), unless the specified ee is already in the template
-     - use role to many-related labels (if any present), unless the specified label is already in the template
-     - use role to many-related instance groups (if any present), unless the specified instance group is already in the template
```
These two are not use-level access.

AlanCoding committed:
Test and fix WFJT node RBAC bug

AlanCoding committed:
Fix a number of bugs from early testing of new prompts

recognize not-provided instance group list on launch, avoiding bug where it fell back to default

fix bug where timeout field was saved on WFJT nodes after creating approval node

remove labels from schedule serializer summary_fields

remove unnecessary prefetch of credentials from WFJT node queryset
